andela/boondocks-bn-frontend
# #169257345 Users should be able to Login to Barefoot Nomad

#### What does this PR do?
This PR implements the login page that allows users to log in to our application

#### Description of Task to be completed?
- [x] create login view
- [x] create login component with an email field, password field, Login button, Forgot password link, Signup link 
- [x] add login functionality; sending a post request to login endpoint with response returned
- [x] add form validation: valid required email; required password 
- [x] handle failed login response with the message rendered using react-toastify
- [x] add CSS styling to form using SASS to match the mockup
- [x] handle successful login: redirect to User Profile
- [x] make sure design is mobile responsive
- [x] add tests for the feature; at least 80% coverage

#### How should this be manually tested?
- clone this repo and then add the following variable to your './.env' file:
  `API_URL=https://boondocks-bn-backend-staging.herokuapp.com`
- navigate to the project folder using Terminal
- install dependencies using `npm install`
- run tests using `npm test`
- start the App using 'npm run watch' and it will be rendered on your browser
- signup onto the App using the signup endpoint '/signup', upon successful registration you will be redirected to the login page '/login'
- attempt to log in to your account with wrong credentials, you will receive an error message
- attempt to login with correct credentials, you will be redirected to the profile page (it will be rendered as a route not found as that feature has not yet been merged)
- while on the browser, open developer tools, on the console, run `document.cookie` to see the cookie that was stored on successful login
- while still on the console, also enter `localStorage.bn_user_data` to see some basic user data stored in localStorage on successful login as well

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#169257345](https://www.pivotaltracker.com/story/show/169257345)

#### Screenshots (if appropriate)
<img width="1436" alt="Screen Shot 2020-01-07 at 06 06 57" src="https://user-images.githubusercontent.com/17108099/71867748-20f11980-3114-11ea-9b42-a09abb79da81.png">
<img width="1438" alt="Screen Shot 2020-01-07 at 06 07 17" src="https://user-images.githubusercontent.com/17108099/71867757-2c444500-3114-11ea-8a50-34c912903a34.png">

#### Questions:
N/A
